tools/report-parser/report-parser.js

```diff
@@ -0,0 +1,37 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import * as fs from "fs";
+import * as util from "util";
+import * as path from "path";
+
+var directory = process.argv.slice(2)[0];
+
+let files  = [];
+
+const getFilesRecursively = (directory) => {
+    const filesInDirectory = fs.readdirSync(directory);
+    for (const file of filesInDirectory) {
+      const absolute = path.join(directory, file);
+      if (fs.statSync(absolute).isDirectory()) {
+          getFilesRecursively(absolute);
+      } else if (/junit-report.json$/.test(absolute)) {
```

Comment:
it's usually best practice to separate policy (the decision to test for `junit-report.json` would be such an example) from generic helpers like "get files recursively".

I'm also a big fan of writing recursive enumeration of a tree structure using [generators](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/function*). basic idea is you do
```js
function* enumerateSubtree(node) {
    yield node;
    for (const child of node.children) {
        yield* enumerateSubtree(child)
    }
}

```

Reply:
Yes, separating policy is usually good way forward. In his case here though it may be fairly small piece of logic and not much benefit of separating policy or need to reuse helpers. 
 
With recursion it's always good to consider the depth of the recursion (as it has stack implications), but in this case this seems to be length of the path which should not be too long. Although don't like names like this `getFilesRecursively` :) as the function name implies implementation details which may change.